Tests/test_integration.py

```diff
@@ -407,6 +407,63 @@ def __test_integration_instance(client, module_instance, prints_manager, thread_
     return success, failure_message
 
 
+def __set_server_keys(client, prints_manager, integration_params, integration_name):
+    if 'server_keys' not in integration_params:
+        return
+
+    prints_manager.add_print_job(f'Setting server keys for integration: {integration_name}',
+                                 print_color, 0, LOG_COLORS.GREEN)
+
+    data = {
+        'data': {},
+        'version': -1
+    }
+
+    for key, value in integration_params.get('server_keys').items():
+        data['data'][key] = value
+
+    response_data, status_code, _ = demisto_client.generic_request_func(self=client, path='/system/config',
+                                                                        method='POST', body=data)
+
+    try:
+        result_object = ast.literal_eval(response_data)
+    except ValueError as err:
+        print_error(
+            'failed to parse response from demisto. response is {}.\nError:\n{}'.format(response_data, err))
+        return
+
+    if status_code >= 300 or status_code < 200:
+        message = result_object.get('message', '')
+        msg = "Failed to set server keys " + str(status_code) + '\n' + message
+        print_error(msg)
+
+
+def __delete_integration_instance_if_determined_by_name(client, instance_name, prints_manager, thread_index=0):
+    try:
+        int_resp = demisto_client.generic_request_func(self=client, method='POST',
+                                                       path='/settings/integration/search',
+                                                       body={'size': 1000})
+        int_instances = ast.literal_eval(int_resp[0])
+    except requests.exceptions.RequestException as conn_err:
+        error_message = 'Failed to delete integrations instance, error trying to communicate with demisto server: ' \
+                        '{} '.format(conn_err)
+        prints_manager.add_print_job(error_message, print_error, thread_index)
+        return
+    if int(int_resp[1]) != 200:
```

Comment:
move this if conditional to line 445 - you do not need to pull the int_instances if the response is not 200.